src/Http/Client.php

```diff
@@ -47,12 +46,13 @@ class Client implements ClientInterface
      *
      * @since 2.0.0
      * @param string $dsn Data Source Name
+     *
+     * @throws \InvalidArgumentException
      */
     public function __construct($dsn)
     {
         $this->driver  = new CurlDriver();
-        $this->request = new Request($dsn);
-        $this->request->withHeader('Content-Type', 'application/json');
+        $this->request = (new Request($dsn))->withHeader('Content-Type', 'application/json');
```

Comment:
PSR-7 requests/responses are immutable, and shouldn't be manipulated without considering immutability